src/recipes/browserDownloadUrls.js

```diff
@@ -0,0 +1,88 @@
+                var box = element.getBoundingClientRect();
+                var docEl = document.documentElement;
+                var scrollTop = docEl.scrollTop;
+                var scrollLeft = docEl.scrollLeft;
+                var clientTop = docEl.clientTop;
+                var clientLeft = docEl.clientLeft;
+                var width = box.width;
+                var height = box.height;
+                var top = box.top + scrollTop - clientTop;
+                var left = box.left + scrollLeft - clientLeft;
+                var bottom = top + height;
+                var right = left + width;
+                if (width == 0 || height == 0 || bottom <= 0 || right <= 0) {
+                    return false;
+                }
+            }          
+            
+            //checks for visibility with computed style
+            var elStyle = getComputedStyle(element);
+            if(elStyle.getPropertyValue("display") === "none"){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("opacity") < 0.1){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("transform").includes(" 0,") || elStyle.getPropertyValue("transform").includes(" 0)")){
+                return false;
+            } 
+            else if(elStyle.getPropertyValue("visibility") === "hidden"){
+                return false;
+            }
+
+            // if text is within an iframe that does not appear
+            var elAbove = element;
+            do{
+                if(elAbove.nodeName === "IFRAME"){
+                    if(getComputedStyle(elAbove).getPropertyValue("src") === "" && getComputedStyle(elAbove).getPropertyValue("frameBorder") === 0){
+                        return false;
+                    }
+                }
+                if(elAbove.parentElement !== null){
+                    elAbove = elAbove.parentElement;
+                }
+            } while(elAbove.parentElement !== null);
+            return true;
+        }
+        
+        //tests for browser download urls
+        var linkList = [{url: (new RegExp("http(s)?\\:\\/\\/(\w{0,9}\\.)?google\\.(\\w{0,4})((\\W|\\w)+)?\/chrome", "gi")), name:"Chrome"}, //but not support.google
+        {url: (new RegExp("microsoft\\.(\\w{0,4})\\/((\\W|\\w)+)?(internet-explorer|\\Wie)($|\\W)", "gi")), name:"Internet Explorer"}, //but not answers.
+        {url: (new RegExp("microsoft\\.(\\w{0,4})\\/(\\W|\\w)+?(microsoft-edge)($|\\W)", "gi")), name:"Edge"}, //but not answers.  
+        {url: (new RegExp("http(s)?\\:\\/\\/(\w{0,9}\\.)?(mozilla|getfirefox|firefox)\\.(\\w{0,4})", "gi")), name:"Firefox"}, //but not support.
+        {url: (new RegExp("http(s)?\\:\\/\\/(\\w{0,9}\\.)?apple\\.(\\w{0,4})\\/((\\w|\\W)+)?safari", "gi")), name:"Safari"}, //but not support.
+        {url: (new RegExp("http(s)?\\:\\/\\/(\w{0,9}\\.)?opera\\.(\\w{0,4})", "gi")), name:"Opera"}]; //but not help.
```

Comment:
new RegExp("^...") // start has to be first character
new RegExp("...$) // end has to be last character
/x/ == new RegExp("x")
/http(s)\:\/\/.../ instead of RegExp("http(s)\\:\\/\\/...")

/
\/
\\/